graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/ApiClientImpl.java

```diff
@@ -508,6 +458,60 @@ public T execute() throws APIException, IOException {
             throw new APIException(request, new IllegalStateException("Unhandled error condition in API client"));
         }
 
+        private T handleResponse(Request request, Response response) throws IOException, APIException {
+            // TODO: once we switch to jackson we can take the media type into account automatically
+            final MediaType responseContentType;
+            if (response.getContentType() == null) {
+                responseContentType = MediaType.JSON_UTF_8;
+            } else {
+                responseContentType = MediaType.parse(response.getContentType());
+            }
+
+            if (!responseContentType.is(mediaType.withoutParameters())) {
+                LOG.warn("We said we'd accept {} but got {} back, let's see how that's going to work out...", mediaType, responseContentType);
+            }
+            if (responseClass.equals(String.class)) {
+                return responseClass.cast(response.getResponseBody("UTF-8"));
```

Comment:
`Response#getResponseBody(String)` already returns a `String`. Why cast it again in this case?